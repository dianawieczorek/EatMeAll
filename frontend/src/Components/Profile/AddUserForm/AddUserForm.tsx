import React from 'react';
import {connect} from 'react-redux';
import Button from "../../UI/Button/Button"
import {deleteUsers, setUserName} from "../../../Redux/actions";
import {Dispatch} from "redux";
import styles from "./AddUserForm.module.css"

interface OwnProps {
}

interface OwnState{
    userToAdd: string;
}

type Props = OwnProps & ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>;
type State = OwnState;

class AddUserForm extends React.Component<Props, State> {

    constructor(props:any) {
        super(props);
        this.state = {userToAdd: ''};
    }

    render() {
        return (
            <div>
                <input className={styles.Input} onChange={this.inputChanged} value={this.state.userToAdd}
                       type="text"/>
                <Button onClick={this.addUser} type="submit">dodaj dietożercę</Button>
                <Button onClick={this.deleteUsers} type="submit">skasuj wszystkich dietożerców</Button>
            </div>
        )
    }

    private addUser = () => {
        console.log(this.state)
        this.props.addUser(this.state.userToAdd);
        this.setState({userToAdd: ""})
    };

    private deleteUsers = () => {
        this.props.deleteAll();
    };

    private inputChanged = (e: any) => {
        console.log(this.state)
        console.log(e.target.value)
        this.setState({userToAdd: e.target.value})
        console.log(this.state)
        console.log(e.target.value)
    };
}

const mapStateToProps = () => {
};

const mapDispatchToProps = (dispatch: Dispatch) => {
    return {
        addUser: (aUserName: string) => dispatch(setUserName(aUserName)),
        deleteAll: () => dispatch(deleteUsers()),
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(AddUserForm)