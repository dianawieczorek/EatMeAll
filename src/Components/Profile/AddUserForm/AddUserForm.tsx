import React, {PureComponent, RefObject} from 'react';
import {connect} from 'react-redux';
import Button from "../../UI/Button/Button"
import {deleteUsers, setUserName} from "../../../Redux/actions";
import {Dispatch} from "redux";


interface OwnProps {
}

type Props = OwnProps & ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>;

class AddUserForm extends PureComponent<Props> {
    readonly userNameInput: RefObject<HTMLInputElement>;

    constructor(Props:any) {
        super(Props);
        this.userNameInput = React.createRef();
    }

    render() {
        return (
            <div>
                <input ref={this.userNameInput} type="text"/>
                <Button onClick={this.addUser} type="submit">dodaj dietożercę</Button>
                <Button onClick={this.deleteUsers} type="submit">skasuj wszystkich dietożerców</Button>
            </div>
        )
    }

    private addUser = () => {
        this.props.addUser(this.userNameInput.current!.value)
    }
    private deleteUsers = () => {
        this.props.deleteAll();
    }
}

const mapStateToProps = () => {
};

const mapDispatchToProps = (dispatch: Dispatch) => {
    return {
        addUser: (aUserName: string) => dispatch(setUserName(aUserName)),
        deleteAll: () => dispatch(deleteUsers()),

    }
};

export default connect(mapStateToProps, mapDispatchToProps)(AddUserForm);
