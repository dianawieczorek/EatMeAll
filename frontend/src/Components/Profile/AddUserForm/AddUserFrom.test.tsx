import React from 'react';
import configureStore from 'redux-mock-store';
import {Provider} from "react-redux";
import renderer from 'react-test-renderer';

import AddUserForm from "./AddUserForm";
import {AnyAction, Store} from "redux";

//CONTRACT:
//1. What should be rendered?
// one test input and two buttons add and delete wrapped by div
//
//2. What component pass to his childs as props?
//  methods to add and delete user, methods implementations come from redux
//
//3. How user can affects to component?
// click on any button and change store
//
//4. Component has any cifecycle behaviour?
// --

// Static HTML
const STATIC_HTML = `<div>
//     <input type="text" class="Input"/>
//     <button class="Button undefined">dodaj dietożercę</button>
//     <button class="Button undefined">skasuj wszystkich dietożerców</button>
// </div>`.replace(/(\r\n|\n|\r)/gm, '').replace(/  +/g, '');


const mockStore = configureStore([]);
let mockedStore: any;
let component: any;

beforeEach(() => {
    mockedStore = mockStore({
        myState: 'sample text',
    });

    component = renderer.create(
        <Provider store={mockedStore}>
            <AddUserForm />
        </Provider>
    );
});


describe('My Connected React-Redux Component', () => {
    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
    
    it('should dispatch an action on button click', () => {
    });
});

