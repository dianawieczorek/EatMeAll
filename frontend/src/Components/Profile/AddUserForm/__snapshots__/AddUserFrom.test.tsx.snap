// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div>
  <input
    className="Input"
    type="text"
  />
  <button
    className="Button undefined"
    onClick={[Function]}
  >
    dodaj dietożercę
  </button>
  <button
    className="Button undefined"
    onClick={[Function]}
  >
    skasuj wszystkich dietożerców
  </button>
</div>
`;
